src/Plugin/migrate/process/VideoVttSource.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Drupal\custom_peertube_migration\Plugin\migrate\process;
+if (!defined('PCDM_TRANS_URI')) {
+        define('PCDM_TRANS_URI', 'http://pcdm.org/use#Transcript');
+        }
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+use Drupal\migrate\MigrateSkipRowException;
+use Drupal\media\Entity\Media;
+use Drupal\file\Entity\File;
+use Drupal\Core\File\FileRepositoryInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\File\FileExists;
+use Drupal\Core\File\FileSystem;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * pull video link from peertube based on media node from drupal
+ *
+ * @MigrateProcessPlugin(
+ *   id = "video_vtt_source"
+ * )
+ */
+
+class VideoVttSource extends ProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+
+    //get peertube api prefix as the pattern to match
+    $uri_prefix =\Drupal::config('custom_peertube_migration.settings')->get('base_uri'); 
+    $pattern = trim($uri_prefix) . "/w/";
+    
+    //get source parameters
+    $videoUrl = $value[0];
+    $parent_repository_item_id = $value[1]; 
+    $video_name = $value[2]; 
+    
+     if ( stripos($videoUrl, $pattern) !== 0) {
+	\Drupal::logger('custom_peertube_migration')->info('Video URL Not matched on record: @data', ['@data' => $video_name]);
+	throw new MigrateSkipRowException('Media Video Source URL does not match peertube pattern');
+	} else {
+    		$videoId = basename(substr($videoUrl, strlen($pattern))); //only take the lastpart of videoUrl
+	        
+		//step2.  handle video captions migration                                                                                                        
+
+		//construct params for vtt
+		$vtt_array = [
+		'prefix' => $uri_prefix,
+		'video_name'=> $video_name,
+		'video_id' => $videoId,
+		'repo_item_id' => $parent_repository_item_id ?? '' 
+		];
+		$this->viedoCaptions_handler($vtt_array); 
+	}
+}
+
+/** 
+ * Retrieve Media Usage from Taxonomy Terms. e.g. Service File, Thumbnail Image, Extracted Text 
+ */
+protected function getMediaUseTerm(string $uri) {
+        $term_query = \Drupal::entityQuery('taxonomy_term')                                                                                             
```

Comment:
Watch for changes in indentation here and elsewhere.  I see a mix of 2 spaces, 4 spaces, and 8 spaces.